core/src/main/scala/kafka/log/Log.scala

```diff
@@ -444,6 +457,21 @@ class Log(@volatile var dir: File,
     }
   }
 
+  /*
+   * Increment the log start offset if the provided offset is larger.
+   * And purge old segments whose largest offset < log start offset.
+   */
+  def maybeIncrementLogStartOffsetAndPurge(offset: Long) {
+    // We don't have to write the log start offset to log-start-offset-checkpoint immediately.
+    // The purgeOffset may be lost only if all replicas of this broker are shutdown
```

Comment:
It was not accurate. I have updated the comment to replace `all replicas of this broker` to `in-sync replicas of this broker`